app/knights/knight.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+        self.armour = []
+        self.weapon = None
+        self.potion = None
+
+    def apply_armour(self, armour: list[dict[str, int]]) -> None:
+        for part in armour:
+            if "protection" in part:
+                self.protection += part["protection"]
+
+    def apply_weapon(self, weapon: dict[str, int]) -> None:
+        if "power" in weapon:
+            self.power += weapon["power"]
+
+    def apply_potion(self, potion: dict[str, dict[str, int]]) -> None:
+        if potion is not None and "effect" in potion:
+            effect = potion["effect"]
+            if "hp" in effect:
```

Comment:
Use for loop and `getattr` and `setattr` functions to apply effects, don't repeat yourself